default_dial_option_server_option_test.go

```diff
@@ -74,6 +75,29 @@ func (s) TestDisableGlobalOptions(t *testing.T) {
 	internal.ClearGlobalDialOptions()
 }
 
+type testLateApplyDialOption struct{}
+
+func (do *testLateApplyDialOption) DialOption(parsedTarget url.URL) DialOption {
+	if parsedTarget.Scheme == "passthrough" {
+		return WithTransportCredentials(insecure.NewCredentials()) // credentials provided, should pass NewClient.
+	}
+	return WithReadBufferSize(10) // no credentials, should fail NewClient
+}
+
+func (s) TestGlobalLateApplyDialOption(t *testing.T) {
+	internal.AddGlobalLateApplyDialOptions.(func(opt ...LateApplyDialOption))(&testLateApplyDialOption{}) // Do I need a clear?
+	noTSecStr := "no transport security set"
+	if _, err := NewClient("fake"); !strings.Contains(fmt.Sprint(err), noTSecStr) {
```

Comment:
An explicit `dns:///` would be helpful here.

Reply:
Good point. Done.